jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ShutdownMonitor.java

```diff
@@ -13,470 +13,318 @@
 import org.eclipse.jetty.util.component.LifeCycle;
 import org.eclipse.jetty.util.thread.AutoLock;
-import org.eclipse.jetty.util.thread.ShutdownThread;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * Shutdown/Stop Monitor thread.
+ * Shutdown Monitor.
+ *
  * <p>
- * This thread listens on the host/port specified by the STOP.HOST/STOP.PORT
- * system parameter (defaults to 127.0.0.1/-1 for not listening) for request
- * authenticated with the key given by the STOP.KEY system parameter
- * for admin requests.
+ *     This is a singleton that is only valid when the following System Properties are defined.
+ * </p>
+ *
+ * <dl>
+ * <dt>{@code STOP.HOST}</dt>
+ * <dd>IP to listen on, defaults to {@code 127.0.0.1}</dd>
+ * <dt>{@code STOP.PORT}</dt>
+ * <dd>Port to listen on, defaults to {@code -1} (or disabled).<br>
+ * (0 will use a port number that is automatically allocated)</dd>
+ * <dt>{@code STOP.KEY}</dt>
+ * <dd>The Key that must be provided to initiate a Shutdown.<br>
+ * Limited to {@link java.nio.charset.StandardCharsets#US_ASCII US_ASCII} charset.<br>
+ * If one is not provided, a generated Key will be created.</dd>
+ * <dt>{@code STOP.EXIT}</dt>
+ * <dd>Boolean to indicate if a {@code System.exit(0)} should occur on successful shutdown,
+ * defaults to {@code true}</dd>
+ * </dl>
+ *
  * <p>
- * If the stop port is set to zero, then a random port is assigned and the
- * port number is printed to stdout.
+ * This starts a {@link java.net.ServerSocket} that listens on the host/port specified by
+ * the configuration, and starts a thread to accept incoming requests.
+ * </p>
+ *
  * <p>
- * Commands "stop" and "status" are currently supported.
+ *     See {@link ShutdownService} for details about commands that can be sent to
+ *     this server.
+ * </p>
+ *
+ * @deprecated Replaced with {@link ShutdownService} component, which is not a singleton.
  */
-public class ShutdownMonitor
+@Deprecated(since = "12.1.0", forRemoval = true)
+public class ShutdownMonitor extends ShutdownService
 {
-    // Implementation of safe lazy init, using Initialization on Demand Holder technique.
-    private static class Holder
+    private static final Logger LOG = LoggerFactory.getLogger(ShutdownMonitor.class);
+    private static final AtomicReference<ShutdownMonitor> INSTANCE = new AtomicReference<>();
+
+    /**
+     * @deprecated No direct replacement, see {@link ShutdownService}, which is not a singleton.
+     */
+    @Deprecated(since = "12.1.0", forRemoval = true)
+    public static ShutdownMonitor getInstance()
     {
-        static ShutdownMonitor instance = new ShutdownMonitor();
+        return INSTANCE.updateAndGet((h) ->
+        {
+            if (h != null)
+                return h;
+            else
+            {
+                LOG.warn("{} is deprecated, and has been replaced with {}", ShutdownMonitor.class.getName(), ShutdownService.class.getName());
+                return new ShutdownMonitor();
```

Comment:
What happens in 12.0.x if the `ShutdownMonitor()` constructor throws?

Try using `-DSTOP.PORT=foo` and see what happens.